packages/uploadthing/src/internal/handler.ts

```diff
@@ -104,12 +117,12 @@ export const buildRequestHandler =
   ): RequestHandler<Args> =>
   ({ req, middlewareArgs }) =>
     pipe(
-      req instanceof Request ? Effect.succeed(req) : req,
+      Effect.isEffect(req) ? req : Effect.succeed(req),
```

Comment:
Effect.isEffect is useful for this kind of thing - using `instanceof` can break if you ship both cjs & esm builds and the user ends up with duplicate modules.